file_reader.go

```diff
@@ -125,9 +125,13 @@ func (f *FileReader) Seek(offset int64, whence int) (int64, error) {
 		return f.offset, fmt.Errorf("Invalid resulting offset: %d", off)
 	}
 
-	f.offset = off
-	f.blockReader = nil
-
+	if f.offset != off {
+		f.offset = off
+		if f.blockReader != nil {
```

Comment:
I think we want this to be a separate conditional, so we close it even if `f.offset == off`